feat: add enhanced relative gap

23Kev opened the pull request:
Now this should be a clean branch/PR for the Relative Gap Improvements feature that only contains the Relative files. 
The Brake Bias change is also included in here.

tariknz commented on src/frontend/components/Standings/hooks/useDriverRelatives.tsx:
```diff
@@ -3,19 +3,30 @@ import {
+  useRelativeGapStore,
 } from '@irdashies/context';
 import { useDriverStandings } from './useDriverPositions';
+import { detectEdgeCases, calculateRelativeGap } from '@irdashies/context/RelativeGapStore';
```
export RelativeGapStore in the index.ts

23Kev commented:
Changes Made:
1. RelativeGapStore.tsx - Added `processPositionUpdates` action
Moved all business logic from the updater hook into the store
Created a single action method that processes all position updates
Follows the established pattern from `LapTimesStore`
2. RelativeGapStoreUpdater.tsx - Complete refactoring
- Replaced `useTelemetryStore((state) => state.telemetry)` with granular hooks:
```
useTelemetryValue('SessionTime')
useTelemetryValue('SessionNum')
useTelemetryValues('CarIdxLapDistPct')
useTelemetryValues('CarIdxLap')
useTelemetryValues('CarIdxLastLapTime')
```
Replaced `parseTrackLength()` function with `useTrackLength()` hook
Removed 18 lines of duplicate track length parsing code
Moved all business logic to store's `processPositionUpdates` actionHook is now minimal (15 lines vs 62 lines) - just passes data to store
